Realm/RLMRealm.mm

```diff
@@ -294,6 +294,17 @@ + (NSString *)writeablePathForFile:(NSString*)fileName
     return [path stringByAppendingPathComponent:fileName];
 }
 
++ (NSString *)writeableTemporaryPathForFile:(NSString*)fileName
+{
+    NSString *path = NSTemporaryDirectory();
+    [[NSFileManager defaultManager] createDirectoryAtPath:path
+                              withIntermediateDirectories:YES
+                                               attributes:nil
+                                                    error:nil];
+    fileName = [NSString stringWithFormat:@"%d-%@", NSProcessInfo.processInfo.processIdentifier, fileName];
```

Comment:
Is there a reason we're doing our own thing rather than using `mkstemp` or similar?